### PR TITLE
King valid_move? & check logic integration - Ready For Review

### PR DESCRIPTION
**Pull request by AlekstheRealist**
- Added not_check? method to valid_move? method

- Added two passing tests to rspec king model

- Anything else we can change? Let us know :)

### CODE REVIEW

**Comment by ruishine** [comment]
should this be reviewed after the check_logic one? aka the check_logic one ready for review?

File: app/models/king.rb [hunk 1]
```diff
@@ -2,13 +2,22 @@ class King < Piece
 
 # valid move for king
   def valid_move?(new_x, new_y)
-    allowed_move?(new_x, new_y) && passes_king_rules?(new_x, new_y)
+    allowed_move?(new_x, new_y) && passes_king_rules?(new_x, new_y) && not_check?(new_x, new_y)
```
Comment (ilyakrasnov) [inline app/models/king.rb]: Nice job preventing moving yourself into check 😉

**Comment by ilyakrasnov** [comment]
👍

**Comment by lindyrp** [comment]
This all looks good to me - tests look great! Thumbs up here.

**Pull request merged**